kani-driver/src/call_cargo.rs

```diff
@@ -181,22 +185,81 @@ impl VerificationTarget {
 }
 
 /// Extract the targets inside a package.
+///
 /// If `--tests` is given, the list of targets will include any integration tests.
+///
+/// We use the `target.kind` as documented here. Note that `kind` for library will
+/// match the `crate-type`, despite them not being explicitly listed in the documentation:
+/// <https://docs.rs/cargo_metadata/0.15.0/cargo_metadata/struct.Target.html#structfield.kind>
+///
+/// The documentation for `crate-type` explicitly states that the only time `kind` and
+/// `crate-type` differs is for examples.
+/// <https://docs.rs/cargo_metadata/0.15.0/cargo_metadata/struct.Target.html#structfield.crate_types>
 fn package_targets(args: &KaniArgs, package: &Package) -> Vec<VerificationTarget> {
-    package
+    let mut ignored_libs = vec![];
+    let mut ignored_tests = vec![];
+    let mut ignored_unsupported = vec![];
+    let verification_targets = package
         .targets
         .iter()
         .filter_map(|target| {
-            debug!(name=?package.name, target=?target.name, kind=?target.kind, "package_targets");
+            debug!(name=?package.name, target=?target.name, kind=?target.kind, crate_type=?target
+                .crate_types,
+                "package_targets");
             if target.kind.contains(&String::from("bin")) {
+                // Binary targets.
                 Some(VerificationTarget::Bin(target.name.clone()))
-            } else if target.kind.contains(&String::from("lib")) {
-                Some(VerificationTarget::Lib)
-            } else if target.kind.contains(&String::from("test")) && args.tests {
-                Some(VerificationTarget::Test(target.name.clone()))
+            } else if target.kind.contains(&String::from("lib"))
+                || target.kind.contains(&String::from("rlib"))
+            {
+                // Lib targets.
+                if target.kind.iter().any(|kind| {
+                    matches!(kind.as_str(), "cdylib" | "dylib" | "staticlib" | "proc-macro")
```

Comment:
Can we define these strings as constants?

Reply:
sure